release builds failing due to publish-docker.sh script

during our releases builds yesterday, the build failed twice during the docker build:

https://travis-ci.com/github/pulumi/pulumi/jobs/332220673#L10127

It seems like the pulumi test runtime isn't building the binary correctly:

https://github.com/pulumi/pulumi/blob/master/scripts/build-docker.sh#L36

or it's not being mounted correctly:

https://github.com/pulumi/pulumi/blob/master/scripts/build-docker.sh#L46

This is the relevant error. You're probably right that it's either one of those things. Let me figure out which.

```
10126Successfully installed appdirs-1.4.4 certifi-2020.4.5.1 distlib-0.3.0 filelock-3.0.12 importlib-metadata-1.6.0 pipenv-2018.11.26 virtualenv-20.0.20 virtualenv-clone-0.5.4 zipp-3.1.0
10127/bin/bash: /src/pulumi-test-containers: No such file or directory
10128Makefile:76: recipe for target 'publish_packages' failed
```

--- a/scripts/build-docker.sh
+++ b/scripts/build-docker.sh
@@ -32,15 +32,16 @@ test_containers() {
     echo_header "Executing container runtime tests"
 
     # Run the container tests, note that we also build the binaries into /tmp for the next step.
+    TEST_BIN_DIRECTORY=$(mktemp -d)
     pushd ${ROOT}/tests
-    GOOS=linux go test -c -o /tmp/pulumi-test-containers ${ROOT}/tests/containers/...
+    GOOS=linux go test -c -o ${TEST_BIN_DIRECTORY} ${ROOT}/tests/containers/...
     popd
 
     # Run tests _within_ the "pulumi" container, ensuring that the CLI is installed
     # and working correctly.
     docker run -e RUN_CONTAINER_TESTS=true \
         -e PULUMI_ACCESS_TOKEN=${PULUMI_ACCESS_TOKEN} \
-        --volume /tmp:/src \
+        --volume ${TEST_BIN_DIRECTORY}:/src \
         --entrypoint /bin/bash \
         pulumi/pulumi:latest \
         -c "pip install pipenv && /src/pulumi-test-containers -test.parallel=1 -test.v -test.run TestPulumiDockerImage"

